Content/WorldGeneration/ChestLootSystem.cs

```diff
@@ -0,0 +1,246 @@
+
+                        if (itemInfoList.Count > 0)
+                        {
+                            // Selects a random exclusive item to be placed in the chest
+                            ChestLootInfo exclusiveitemInfo = WorldGen.genRand.Next(itemInfoList);
+
+                            if (WorldGen.genRand.NextFloat() < exclusiveitemInfo.chance)
+                                AddChestItem(exclusiveitemInfo, chest);
+                        }
+                    }
+
+                    // If this is a wooden dungeon chest is not checked for non-exclusive items
+                    {
+                        // Gets all valid items for this chest type plus the all chest type and then shuffles it.
+                        List<ChestLootInfo> itemInfoList = new(RegionLootInfo[region]);
+                        itemInfoList.AddRange(RegionLootInfo[ChestRegionFlags.All]);
+                        itemInfoList = itemInfoList.OrderBy(x => WorldGen.genRand.Next()).ToList();
+
+                        if(itemInfoList.Count > 0)
+                            foreach (ChestLootInfo itemInfo in itemInfoList)
+                                if (WorldGen.genRand.NextFloat() < itemInfo.chance)
+                                    AddChestItem(itemInfo, chest);
+                    }
+
+                    //if (WorldGen.genRand.NextFloat() < displayCaseChance && IsDisplayCaseReplaceable(tile.TileFrameX))
+                    //{
+                    //    PlaceDisplayCaseOn(chest);
+                    //
+                    //    // Continues because we don't want the code after this to touch a chest that is no longer accessible.
+                    //    continue;
+                    //}
+                }
+            }
+        }
+
+        private static Item SetupItem(int type, int stack, bool isRelic)
+        {
+            Item Item = new Item(type, stack);
+
+            Item.GetGlobalItem<RelicItem>().isRelic = isRelic;
+            Item.Prefix(ItemLoader.ChoosePrefix(Item, Main.rand));
+
+            return Item;
+        }
+        private static void AddChestItem(ChestLootInfo info, Chest chest)
+        {
+            int stack = WorldGen.genRand.Next(info.stackRange.Item1, info.stackRange.Item2 + 1);
+            int slot = info.slotIndex;
+
+            // Finds first open slot
+            if (slot == -1)
+                for (int g = 0; g < chest.item.Length; g++)
+                    if (chest.item[g].IsAir)
+                    {
+                        slot = g;
+                        break;
+                    }
+
+            // Slot is checked in case no open slot was found, Stack is checked in case the minimum was zero
+            if (slot != -1 && stack > 0)
+                chest.item[slot] = SetupItem(info.itemType, stack, false);//isRelic option is unused
+        }
+        //private static void PlaceDisplayCaseOn(Chest chest)
```

Comment:
This should likely be moved to it's own generation pass. Lets create a seperate ticket for this.